### PR TITLE
task/DES-370

### PR DESCRIPTION
**Pull request by jrfreez**
Went through design design safe directories adding .then and error to $http calls wherever it was absent in .js files. Most were in the services directories but there were others.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+package-lock.json
```
Comment (xirdneh) [inline .gitignore]: We have to track these files to make sure that all the JS files are the same everywhere.

**Comment by jrfreez** [comment]
DES-370 should be resolved now (and the gitignore file as well).

Joshua

**Comment by xirdneh** [comment]
@jrfreez I had to revert this PR because there's a piece missing. We need to make sure that the services that were updated are being used in the correct way.

**Comment by jrfreez** [comment]
What do I need to do with it now?
Joshua

**Pull request merged**